Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm

```diff
@@ -110,11 +113,22 @@ - (WKWebView *)webView
         [_webView _setAutomaticallyAdjustsContentInsets:NO];
         [_webView _setUseSystemAppearance:YES];
         [_webView setAutoresizingMask:NSViewWidthSizable | NSViewHeightSizable];
+
+        [_webView _setObscuredContentInsets:self.webView.safeAreaInsets immediate:NO];
```

Comment:
Are you sure this is safe to do? Clients can set other insets beyond the safe area insets, for example if they are going to show a banner over part of the web view. In those cases it seems like the client's inset still needs to be honored...

Reply:
I think this is quite safe to do. A view's `safeAreaInsets` should already honor any custom insets applied on its parent, including any `additionalSafeAreaInsets` or constraints it has. Sure, `safeAreaInsets` doesn't account for customizations on the view itself, but we as the owner of the inspector view know that it doesn't have any custom insets on it.

Reply:
We chatted offline. My key misunderstanding was forgetting that `_webView` is the Web Inspector's web view, not the web view that a client like Safari created.